var/spack/repos/builtin/packages/libyogrt/package.py

```diff
@@ -58,3 +58,19 @@ def configure_args(self):
             args.append('--enable-static=yes')
 
         return args
+
+    def install(self, spec, prefix):
+
+        # Run standard Autotools install first, then create a yogrt.conf file for the selected scheduler
+
+        AutotoolsPackage.install(self, spec, prefix)
+        etcpath = os.path.join(prefix,"etc")
+        if not os.path.isdir(etcpath):
+            mode = 0o755
+            os.mkdir(etcpath, mode)
+            
+        f = open(os.path.join(etcpath,"yogrt.conf"),"w+")
+        f.write("backend=%s\n" % self.spec.variants['scheduler'].value)
+        f.close()
```

Comment:
Don't know if Spack has a preference, but I've see file handling suggested as using this pattern, which relies on python to cleanup in case of an exception:
```suggestion
        with open(os.path.join(etcpath,"yogrt.conf"),"w+") as f:
            f.write("backend=%s\n" % self.spec.variants['scheduler'].value)
```

Reply:
Looks good. I write my python code in C, and don't know the idioms well.